src/mongo_logger.ts

```diff
@@ -329,15 +358,36 @@ function attachCommandFields(
 
 function attachConnectionFields(
   log: Record<string, any>,
-  connectionPoolEvent: ConnectionPoolMonitoringEvent
+  connectionOrSDAMEvent: ConnectionPoolMonitoringEvent | ServerOpeningEvent | ServerClosedEvent
 ) {
-  const { host, port } = HostAddress.fromString(connectionPoolEvent.address).toHostPort();
+  const { host, port } = HostAddress.fromString(connectionOrSDAMEvent.address).toHostPort();
   log.serverHost = host;
   log.serverPort = port;
 
   return log;
 }
 
+function attachSDAMFields(log: Record<string, any>, sdamEvent: ServerDiscoveryAndMonitoringEvent) {
+  log.topologyId = sdamEvent.topologyId;
+  return log;
+}
+
+function attachServerHeartbeatFields(
+  log: Record<string, any>,
+  serverHeartbeatEvent:
+    | ServerHeartbeatFailedEvent
+    | ServerHeartbeatStartedEvent
+    | ServerHeartbeatSucceededEvent
+) {
+  const { awaited, connectionId } = serverHeartbeatEvent;
+  log.awaited = awaited;
+  log.driverConnectionId = 1;
```

Comment:
should this have been assigned to connectionId?

Reply:
Yes, It should have.. it's a remanent of personal debugging, I'll change it.